pilot/pkg/networking/core/cluster_builder.go

```diff
@@ -352,18 +351,20 @@ func (cb *ClusterBuilder) buildCluster(name string, discoveryType cluster.Cluste
 	if direction == model.TrafficDirectionOutbound {
 		// If stat name is configured, build the alternate stats name.
 		if len(cb.req.Push.Mesh.OutboundClusterStatName) != 0 {
-			statPrefix := telemetry.BuildStatPrefix(cb.req.Push.Mesh.OutboundClusterStatName, string(service.Hostname), subset, port, 0, &service.Attributes)
-			if util.IsIstioVersionGE123(cb.proxyVersion) && statPrefix[len(statPrefix)-1:] != constants.StatPrefixDelimiter {
-				statPrefix += constants.StatPrefixDelimiter
-			}
-
-			ec.cluster.AltStatName = statPrefix
+			ec.cluster.AltStatName = telemetry.BuildStatPrefix(cb.req.Push.Mesh.OutboundClusterStatName, string(service.Hostname), subset, port, 0, &service.Attributes)
 		}
 	}
 
 	return ec
 }
 
+func (cb *ClusterBuilder) defaultAltStatName(clusterName string) string {
+	if util.IsIstioVersionGE123(cb.proxyVersion) {
+		return clusterName + constants.StatPrefixDelimiter
+	}
+	return ""
```

Comment:
Why do we return an empty string here? I may not have all of the context but the name implies to me that a default name should always get returned. Or is it that the default name is just this logic:
```go
if util.IsIstioVersionGE123(cb.proxyVersion) {
	return clusterName + constants.StatPrefixDelimiter
}
```

But what if it's not IsIstioVersionGE123, what happens when MeshConfig option does not overwrite the alt stat name as described in the comment on line 300 `// Build default alt stat name - This may be overwritten by the MeshConfig options`. Is it just an empty string or does something end up updating it regardless or does it not matter?